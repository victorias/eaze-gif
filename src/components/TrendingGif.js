// @flow

import React, { PureComponent } from 'react';
import styled from 'styled-components';

import Modal from './Modal';

type Props = {
  id: string,
  stillUrl: string,
  previewUrl: string,
  originalUrl: string,
};

type State = {
  isHovering: boolean,
  isModalOpen: boolean,
};

const Container = styled.div`
  display: flex;
  flex-direction: row;
  flex: 0 0 auto;
`;

const Img = styled.img`flex: 0 1 auto;`;

class TrendingGif extends PureComponent<Props, State> {
  state = {
    isHovering: false,
    isModalOpen: false,
  };

  onMouseOver = () => {
    this.setState({
      isHovering: true,
    });
  };

  onMouseOut = () => {
    this.setState({
      isHovering: false,
    });
  };

  render() {
    const { props, state } = this;
    return (
      <Container
        onMouseOver={this.onMouseOver}
        onMouseOut={this.onMouseOut}
        onClick={() => {
          this.setState({ isModalOpen: true });
        }}
      >
        {state.isHovering ? (
          <Img src={props.previewUrl} alt="gif" />
        ) : (
          <Img src={props.stillUrl} alt="still gif" />
        )}

        {state.isModalOpen && (
          <Modal
            isOpen={state.isModalOpen}
            onRequestClose={() => this.setState({ isModalOpen: false })}
          >
            <Img src={props.originalUrl} alt="original gif" />
          </Modal>
        )}
      </Container>
    );
  }
}

export default TrendingGif;
