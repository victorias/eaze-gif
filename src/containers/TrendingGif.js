// @flow

import { connect } from 'react-redux';
import TrendingGif from '../components/TrendingGif';

import * as selectors from '../redux/gifs/selectors';

const mapStateToProps = (state, props) => ({
  stillUrl: selectors.getTrendingStillFromId(props.id)(state),
  url: selectors.getTrendingGifFromId(props.id)(state),
});

export default connect(mapStateToProps)(TrendingGif);
