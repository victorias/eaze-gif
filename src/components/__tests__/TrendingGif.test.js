// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import TrendingGif from '../TrendingGif';

describe('TrendingGif', () => {
  const baseGif = {
    url: 'gif-url',
    width: '100',
    height: '100',
    size: '100',
  };

  const fixedHeight = {
    ...baseGif,
    mp4: 'mp4',
    mp4_size: 'mp4_size',
    webp: 'webp',
    webp_size: 'webp_size',
  };

  const original = {
    ...fixedHeight,
    frames: '100',
    hash: 'hash',
  };

  it('renders', () => {
    const tree = renderer.create(
      <TrendingGif
        id=""
        original={original}
        still={baseGif}
        preview={fixedHeight}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
