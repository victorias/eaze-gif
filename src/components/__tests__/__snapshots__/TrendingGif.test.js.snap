// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingGif renders 1`] = `
<div
  className="sc-bdVaJa gtnNos"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <img
    alt="still"
    className="sc-ifAKCX bTAHyA"
    height="100"
    src="gif-url"
    width="100"
  />
</div>
`;
